### PR TITLE
fix cost function

### PR DESCRIPTION
**Pull request by mynameisvinn**
your cost function for linear regression should be average loss, not total loss - youll see much faster convergence during training :-)

### CODE REVIEW

**Comment by jostmey** [comment]
If you don't divide the the error by the batch size, you can get away with a smaller learning rate. So if you take the mean, the learning rate will need to go up to compensate. That line of code will need to be changed.

If we use the mean error from the batch in "tensor.py", we will also need to change the corresponding code in serial.py to match.
